tests/operator/helper.go

```diff
@@ -51,6 +54,97 @@ func SplitCsv(csv string) CsvResult {
 	return result
 }
 
+func getAllPodsBy(namespace string, allPods []*provider.Pod) (podsInNamespace []*provider.Pod) {
+	for i := range allPods {
+		pod := allPods[i]
+		if pod.Namespace == namespace {
+			podsInNamespace = append(podsInNamespace, pod)
+		}
+	}
+	return podsInNamespace
+}
+
+func getAllOperatorsBy(namespace string, allOperators []*provider.Operator) (operatorsInNamespace []*provider.Operator) {
+	for _, operator := range allOperators {
+		if operator.Csv.Namespace == namespace || len(operator.TargetNamespaces) == 0 {
+			operatorsInNamespace = append(operatorsInNamespace, operator)
+		}
+	}
+	return operatorsInNamespace
+}
+
+func isSingleNamespacedOperator(operator *provider.Operator) bool {
+	return len(operator.TargetNamespaces) == 1 && operator.Namespace != operator.TargetNamespaces[0]
+}
+
+func isMultiNamespacedOperator(operator *provider.Operator) bool {
+	return len(operator.TargetNamespaces) > 1 && !stringhelper.StringInSlice(operator.TargetNamespaces, operator.Namespace, false)
+}
+
+func checkIfOperatorUnderTest(operator *provider.Operator) bool {
+	for _, testOperator := range env.Operators {
+		if testOperator.Name == operator.Name && testOperator.Namespace == operator.Namespace {
+			return true
+		}
+	}
+
+	return false
+}
+
+func checkValidOperatorInstallation(namespace string) (isDedicatedOperatorNamespace bool, singleOrMultiNamespaceOperators,
+	nonSingleOrMultiNamespaceOperators, csvsFoundButNotInOperatorInstallationNamespace, operatorsFoundButNotUnderTest, podsNotBelongingToOperators []string) {
+	// 1. operator installation checks
+
+	for _, operator := range getAllOperatorsBy(namespace, env.AllOperators) {
+		if namespace == operator.Csv.Annotations["olm.operatorNamespace"] {
+			if checkIfOperatorUnderTest(operator) {
+				if isSingleNamespacedOperator(operator) || isMultiNamespacedOperator(operator) {
+					singleOrMultiNamespaceOperators = append(singleOrMultiNamespaceOperators, operator.Name)
+				} else {
+					nonSingleOrMultiNamespaceOperators = append(nonSingleOrMultiNamespaceOperators, operator.Name)
+				}
+			} else {
+				operatorsFoundButNotUnderTest = append(operatorsFoundButNotUnderTest, operator.Name)
+			}
+		} else {
+			csvsFoundButNotInOperatorInstallationNamespace = append(csvsFoundButNotInOperatorInstallationNamespace, operator.Name)
```

Comment:
csvs from All namespace operators should be fine since they are ok from requirements, unless they are actually installed in the tenant namespace. I would add:
} else if len(operator.TargetNamespaces) !=0  {